modules/dists/BlockDist.chpl

```diff
@@ -1535,6 +1535,14 @@ where this.sparseLayoutType == unmanaged DefaultDist &&
   return true;
 }
 
+// Block1 <=> Block2 
+proc BlockArr.doiSwap(arr) {
+  coforall (loc, localArr1, localArr2) in zip(Locales,this.locArr, arr.locArr) {
+    localArr1.myElems.doiSwap(localArr2.myElems);
+    localArr1.locRAD <=> localArr2.locRAD;
```

Comment:
Not that important, but you can imagine implementing a `LocBlockArr.doSwap()` that does both these swaps and call it here. But I am just saying this as a different way of implementing the swap.